totalRP3/core/impl/slash.lua

```diff
@@ -46,15 +47,35 @@ function TRP3_API.slash.unregisterCommand(commandID)
 	COMMANDS[commandID] = nil;
 end
 
+
+function TRP3_API.slash.registerConditional(conditionalStructure)

```

Comment:
I imagine the idea going forward is to actually dump API stuff in the AddOn_TotalRP3 table somewhere. Unless this is supposed to be internal-API only, though if it is I'd do away with register/unregister and actually make the conditionals table constant.

Reply:
Also, on the one hand the parameter being a table matches the API convention used by `registerCommand` but I'm wondering if it could be clearer if instead there was a `MacroConditional` class (using Ellyb's class stuff), such that API usage would be more akin to:

```
local condition = AddOn_TotalRP3.slash.MacroConditional("mounted", function(...) --[[ test ]] end);
AddOn_TotalRP3.slash.registerConditional(condition);
```

The aim being to make a bit clearer for users of the API; if they know it takes a specific class since they can look up the possible arguments for the constructor vs. looking at the implementation of this function to determine what magic keys are available.

Reply:
I'm not sure about this, but for future public APIs going in `AddOn_TotalRP3` a documentation of the table would be provided so that people don't have to go look at the implementation.

In general, I've been going with OOP for the new system I create for more clarity and extensibility, but I'm okay with have things that are piggybacking on existing systems to not go with classes. If Connor feels fancy he can make that a proper class, but I won't require it here.